lib/samlr/signature.rb

```diff
@@ -61,7 +61,11 @@ def x509
 
     # Establishes trust that the remote party is who you think
     def verify_fingerprint!
-      fingerprint.compare!(certificate!.fingerprint)
+      if fingerprint.is_a?(FingerprintSHA1)
+        fingerprint.compare!(certificate!.fingerprint_sha1)
```

Comment:
does a sha1 have fingerprint_sha256 ? if not maybe just keep that method as fingerprint ?